components/collector/src/source_collectors/quality_time/completeness.py

```diff
@@ -0,0 +1,84 @@
+"""Quality-time completeness collector."""
+
+from typing import Dict, Set
+from collector_utilities.type import URL
+from source_model import SourceMeasurement, SourceResponses
+from source_model.entity import Entities, Entity
+
+from .base import QualityTimeCollector
+
+
+Measurements = list[dict[str, dict[str, str]]]
+
+
+class QualityTimeCompleteness(QualityTimeCollector):
+    """Collector to get the "completeness" metric from Quality-time."""
+
+    async def _parse_source_responses(self, responses: SourceResponses) -> SourceMeasurement:
+        """Get the metric entities from the responses."""
+        datamodel_response = responses[0]
+        reports_response = responses[1]
+        reports = await self._get_reports(reports_response)
+        datamodel = await datamodel_response.json()
+
+        entity_dict = {}
+        for report in reports:
+            report_title = report["title"]
+            subject_types = self.__get_subject_types(report)
+            possible_metrics = self.__get_possible_metrics(datamodel, subject_types)
+            actual_metrics_types = self.__get_actual_metric_types(report)
+
+            for metric_type, entity in possible_metrics.items():
+                if metric_type not in actual_metrics_types:
+                    if metric_type not in entity_dict:
+                        entity["reports"] = report_title
+                        entity_dict[metric_type] = entity
+                    else:
+                        entity_dict[metric_type]["reports"] = entity_dict[metric_type]["reports"] + f", {report_title}"
+
+        entities = Entities(list(entity_dict.values()))
+        return SourceMeasurement(total=str(len(possible_metrics)), entities=entities)
+
+    async def _get_source_responses(self, *urls: URL) -> SourceResponses:
+        """Get responses for reports and the datamodel."""
+        api_url = urls[0]
+        datamodel_url = URL(f"{api_url}/api/v3/datamodel")
+        reports_url = URL(f"{api_url}/api/v3/reports")
```

Comment:
Maybe the API version "v3" should be a class attribute of base.QualityTimeCollector?